packages/react/src/PricingOptions/PricingOptions.tsx

```diff
@@ -382,35 +396,50 @@ type ResponsiveExpandableProps = {
   wide: boolean
 }
 
-const defaultExpanded: ExpandedProp = {
-  narrow: false,
-  regular: true,
-  wide: true,
-}
-
 type ValidFeatureListChildren = {
   Heading: React.ReactElement<PricingOptionsFeatureListGroupHeadingProps> | null
   Items: React.ReactElement<PricingOptionsFeatureListItemProps>[]
 }[]
 
 const PricingOptionsFeatureList = forwardRef<HTMLDivElement, PricingOptionsFeatureListProps>(
-  (
-    {
-      children,
-      className,
-      'data-testid': testId,
-      hasDivider = true,
-      expanded = defaultExpanded,
-      accordionAs = 'h4',
-      ...rest
-    },
-    ref,
-  ) => {
-    const runOnce = React.useRef(false)
-    const [isAccordionOpen, setIsAccordionOpen] = React.useState<boolean>(false)
-    const {allFeatureListsExpanded, updateFeatureListExpanded} = usePricingOptions()
+  ({children, className, 'data-testid': testId, hasDivider = true, expanded, accordionAs = 'h4', ...rest}, ref) => {
+    const {
+      allFeatureListsExpanded,
+      updateFeatureListExpanded,
+      featureListUserInteracted,
+      setFeatureListUserInteracted,
+    } = usePricingOptions()
+    const {isMedium: isRegular, isXLarge: isWide, isSmall: isNarrow} = useWindowSize()
+
+    React.useEffect(() => {
+      if (expanded === undefined || typeof expanded === 'object') {
+        setFeatureListUserInteracted(false)
+      }
```

Comment:
Could you help me understand the purpose of this logic please?

I understand the mechanics of the logic (if the viewport size changes, or if the value of the `expanded` prop changes to `undefined` or any object, then we set `featureListUserInteracted` back to false) but I don't understand _why_ we need this check.